src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1373,6 +1373,12 @@ public override Task<T> GetFieldValueAsync<T>(int ordinal, CancellationToken can
             if (cancellationToken.IsCancellationRequested)
                 return Task.FromCanceled<T>(cancellationToken);
 
+            if (typeof(T) == typeof(Stream))
+                return Unsafe.As<Task<T>>(GetStreamAsync(ordinal, cancellationToken));
```

Comment:
Any reason not to do like you did in the sync version?

```suggestion
                return (Task<T>)(object)GetStreamAsync(ordinal, cancellationToken);
```

Let's avoid scary Unsafe casts unless they're necessary...

Reply:
>  GetStream contains a check that the handler is a ByteHandler, that check needs to be removed.

Tracked by #3152 which awaits you.

> Any reason not to do like you did in the sync version?

Because there is a constraint in `Unsafe.As<T>(object)` which doesn't allow to use it if `T` isn't a class. Could use a dirty magic to avoid this restriction without any reflection, but not happy with it. But here we have a task (which is a reference type) and we sure that `T` is the right type because of the condition one line above.